src/dash-table/components/ControlledTable/fragments/TableTooltip.tsx

```diff
@@ -52,71 +52,52 @@ export default class TableTooltip extends Component<ITooltipProps, IState> {
 
         const el = ReactDOM.findDOMNode(this.refs.tooltip) as any;
 
-        const {positionalParent, parent} = tooltipHelper(el, cell);
+        const positionalParent = getPositionalParent(el);
 
-        if (!positionalParent || !parent || !el) {
+        if (!positionalParent || !cell || !el) {
             return;
         }
 
         const positionalBounds = positionalParent.getBoundingClientRect();
-        const parentBounds = parent.getBoundingClientRect();
+        const parentBounds = cell.getBoundingClientRect();
 
         const {clientWidth: elWidth, clientHeight: elHeight} = el;
+
         const elAnchorHeight = Math.max(
             parseFloat(getComputedStyle(el, ':before').borderWidth || '0'),
             parseFloat(getComputedStyle(el, ':after').borderWidth || '0')
         );
 
         const leftCorrection = (parentBounds.width - elWidth) / 2;
+
         let left =
             parentBounds.left -
             positionalBounds.left +
             positionalParent.scrollLeft +
             leftCorrection;
+
         let top =
             parentBounds.top -
             positionalBounds.top +
             positionalParent.scrollTop +
             parentBounds.height;
 
-        const {
-            scrollLeft: vwLeft,
-            scrollTop: vwTop,
-            clientWidth: vwWidth,
-            clientHeight: vwHeight
-        } = document.body;
-
-        let arrow: Arrow | undefined = Arrow.Top;
+        const leftmost = left + positionalBounds.left;
```

Comment:
Simplify positioning logic. Tooltip position is relative position in relative parent + the relative parent's position.

Make sure the tooltip fits the viewport bounds (document.documentElement) instead of the document body.